### PR TITLE
Fix: ufuncs on records should not be allowed unless overridden

### PR DESCRIPTION
**Pull request by ioanaif**
#1439

### CODE REVIEW

**Comment by ioanaif** [comment]
Now that ufuncs are no longer allowed for records in v2, test `0355-mixins` fails with`Error details: cannot broadcast records in absolute` @jpivarski

**Comment by jpivarski** [comment]
While it's possible that we'll find some tests that assume you can apply ufuncs (like `np.abs`) to generic records and those tests have to be changed or removed, this particular test _should_ still work.

In the definition of Point, a method was defined for `np.abs`:

https://github.com/scikit-hep/awkward/blob/afe5fc166a78b2d062d59811e91a0a0c3fc17dad/tests/test_0355-mixins.py#L18-L20

The rule should be that records can't be used in ufuncs unless a method has been defined for that particular ufunc, as is the case here. That's what `broadcast_and_apply` does; `recursively_apply` should follow the same rule. (That is, `np.add(record, record)` should fail if `record` does not have an implementation of `np.add`, but it should use that implementation if it does.)

File: tests/v2/test_1559_fix_ufuncs_records_1439.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import awkward as ak  # noqa: F401
+import numpy as np  # noqa: F401
+
+to_list = ak._v2.operations.to_list
+
+
+def test_ufuncs_on_records_1439():
+    array = ak._v2.Array([{"x": 1.0, "y": 1.1}, {"x": 2.0, "y": 2.2}])
+    with pytest.raises(ValueError):
+        np.absolute(array)
```
Comment (jpivarski) [inline tests/v2/test_1559_fix_ufuncs_records_1439.py]: It's good that this complains if `np.absolute` is not overloaded, but it should also not complain if `np.absolute` is overloaded. (One more test.)

**Review by jpivarski — APPROVED** [review]
Yes! `recursively_apply` should be just like `broadcast_and_apply` in requiring `behavior` and not allowing records unless the records are overloaded.

Just need a test to be sure that `np.absolute` will work if it is overloaded (like `np.add`).

Once that test has been added and it passes, go ahead and merge.

**Pull request merged**